neqo-transport/src/packet/mod.rs

```diff
@@ -48,41 +49,37 @@ pub use metadata::MetaData;
 
 pub type PacketNumber = u64;
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, FromRepr)]
+#[repr(u8)]
 pub enum PacketType {
-    VersionNegotiation,
-    Initial,
-    Handshake,
-    ZeroRtt,
-    Retry,
+    Initial = 0,
+    ZeroRtt = 1,
+    Handshake = 2,
+    Retry = 3,
     Short,
     OtherVersion,
+    VersionNegotiation,
 }
 
 impl PacketType {
     #[must_use]
     fn from_byte(t: u8, v: Version) -> Self {
         // Version2 adds one to the type, modulo 4
-        match t.wrapping_sub(u8::from(v == Version::Version2)) & 3 {
-            0 => Self::Initial,
-            1 => Self::ZeroRtt,
-            2 => Self::Handshake,
-            3 => Self::Retry,
-            _ => panic!("packet type out of range"),
-        }
+        Self::from_repr(t.wrapping_sub(u8::from(v == Version::Version2)) & 3)
+            .expect("packet type in range")
     }
 
     #[must_use]
     fn to_byte(self, v: Version) -> u8 {
-        let t = match self {
-            Self::Initial => 0,
-            Self::ZeroRtt => 1,
-            Self::Handshake => 2,
-            Self::Retry => 3,
-            _ => panic!("not a long header packet type"),
-        };
+        debug_assert!(
+            matches!(
+                self,
+                Self::Initial | Self::ZeroRtt | Self::Handshake | Self::Retry
+            ),
+            "is a long header packet type"
+        );
         // Version2 adds one to the type, modulo 4
-        (t + u8::from(v == Version::Version2)) & 3
+        (self as u8 + u8::from(v == Version::Version2)) & 3
```

Comment:
Previously on a short header type, this function would panic before this line in release mode. Now it only panics in debug mode, but still executes it in release mode. Is that intended?